File: Selenium/Activities/Activity15.java
import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity15 {

    public static void main(String[] args) {

        ChromeDriver driver = new ChromeDriver();
        
        
        driver.manage().window().maximize();

        driver.get("https://training-support.net/webelements/dynamic-attributes");
        System.out.println("Title of page is : " + driver.getTitle());

        WebElement name = driver.findElement(By.cssSelector("input[id^='full-name']"));
        WebElement email = driver.findElement(By.cssSelector("input[id$='email']"));
        WebElement date = driver.findElement(By.xpath("//input[contains(@name, '-event-date-')]"));
        WebElement additionalDetail = driver.findElement(By.xpath("//textarea[contains(@name, '-additional-details-')]"));
        WebElement button = driver.findElement(By.xpath("//button[text()='Submit']"));
        WebElement success_message = driver.findElement(By.id("action-confirmation"));
        
        name.sendKeys("ritika");
        email.sendKeys("ritika.takke@gmail.com");
        date.sendKeys("2025-06-21");
        additionalDetail.sendKeys("we are testing dynamic page");
        button.click();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
        if(success_message.isDisplayed()) {
        	System.out.println("Success message is dsiplayed");
        }
        
        
        
        driver.quit();
    }
}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
import time

driver = webdriver.Chrome()
driver.maximize_window()

driver.get("https://training-support.net/webelements/dynamic-attributes")
print("Title of page is :", driver.title)

name = driver.find_element(By.CSS_SELECTOR, "input[id^='full-name']")
email = driver.find_element(By.CSS_SELECTOR, "input[id$='email']")
date = driver.find_element(By.XPATH, "//input[contains(@name, '-event-date-')]")
additional_detail = driver.find_element(By.XPATH, "//textarea[contains(@name, '-additional-details-')]")
button = driver.find_element(By.XPATH, "//button[text()='Submit']")
success_message = driver.find_element(By.ID, "action-confirmation")

name.send_keys("ritika")
email.send_keys("ritika.takke@gmail.com")
date.send_keys("2025-06-21")
additional_detail.send_keys("we are testing dynamic page")
button.click()

driver.implicitly_wait(20)

if success_message.is_displayed():
    print("Success message is dsiplayed")

driver.quit()
